listening-platform/src/main/kotlin/com/nikitahohulia/listeningplatform/service/UserServiceImpl.kt

```diff
@@ -88,7 +94,9 @@ class UserServiceImpl(
             .flatMap { (user, publisher) ->
                 user.subscriptions.add(publisher.id!!)
                 userRepository.save(user)
-            }.thenReturn(Unit)
+            }
+            .doOnNext { kafkaUserProducer.sendUserUpdatedEventToKafka(it.toResponse().toProto()) }
+            .thenReturn(Unit)
```

Comment:
A user is not updated here, but deleted. how do consumers of your event know about it?

Reply:
I don't really understand. I checked the code lines you are referring to. They are used to updating the user, more specifically, updating the user by filling out the set of publisher IDs which the user is subscribed to. So it's an update and not a delete. Or I didn't understand you correctly?

Reply:
sorry, my bad